Buildfile: build.xml

javacc-uptodate-check:

javacc-notice:

init:

common.compile-core:

compile-core:

compile-demo:

common.compile-test:

compile-test:

test:
    [junit] Testsuite: org.apache.lucene.index.TestParallelReader
    [junit] Tests run: 2, Failures: 0, Errors: 1, Time elapsed: 1.793 sec

    [junit] Testcase: testMerge(org.apache.lucene.index.TestParallelReader):	Caused an ERROR
    [junit] null
    [junit] java.lang.NullPointerException
    [junit] 	at org.apache.lucene.index.TermInfosReader.get(TermInfosReader.java:140)
    [junit] 	at org.apache.lucene.index.SegmentTermDocs.seek(SegmentTermDocs.java:50)
    [junit] 	at org.apache.lucene.index.IndexReader.termPositions(IndexReader.java:441)
    [junit] 	at org.apache.lucene.index.ParallelReader$ParallelTermPositions.seek(ParallelReader.java:323)
    [junit] 	at org.apache.lucene.index.SegmentMerger.appendPostings(SegmentMerger.java:317)
    [junit] 	at org.apache.lucene.index.SegmentMerger.mergeTermInfo(SegmentMerger.java:288)
    [junit] 	at org.apache.lucene.index.SegmentMerger.mergeTermInfos(SegmentMerger.java:262)
    [junit] 	at org.apache.lucene.index.SegmentMerger.mergeTerms(SegmentMerger.java:226)
    [junit] 	at org.apache.lucene.index.SegmentMerger.merge(SegmentMerger.java:88)
    [junit] 	at org.apache.lucene.index.IndexWriter.addIndexes(IndexWriter.java:599)
    [junit] 	at org.apache.lucene.index.TestParallelReader.testMerge(TestParallelReader.java:63)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)


    [junit] Test org.apache.lucene.index.TestParallelReader FAILED

BUILD FAILED
/Users/skirsch/text/lectures/da/thirdparty/lucene-trunk/common-build.xml:188: Tests failed!

Total time: 9 seconds
Index: src/java/org/apache/lucene/index/ParallelReader.java
===================================================================
--- src/java/org/apache/lucene/index/ParallelReader.java	(revision 191717)
+++ src/java/org/apache/lucene/index/ParallelReader.java	(working copy)
@@ -79,7 +79,7 @@
 
     if (!ignoreStoredFields)
       storedFieldReaders.add(reader);             // add to storedFieldReaders
-
+    readers.add(reader);
   }
 
 
@@ -291,8 +291,12 @@
     }
 
     public void seek(TermEnum termEnum) throws IOException {
+      if (termEnum.term() == null) {
+        termDocs = ((IndexReader)fieldToReader.get(((ParallelTermEnum) termEnum).field)).termDocs(termEnum.term());
+      } else {
       seek(termEnum.term());
     }
+    }
 
     public boolean next() throws IOException { return termDocs.next(); }
 
@@ -314,6 +318,15 @@
     public ParallelTermPositions() {}
     public ParallelTermPositions(Term term) throws IOException { seek(term); }
 
+    public void seek(TermEnum termEnum) throws IOException {
+      if (termEnum.term() == null) {
+        termDocs = ((IndexReader)fieldToReader.get(((ParallelTermEnum) termEnum).field)).termPositions(termEnum.term());
+      } else {
+        seek(termEnum.term());
+      }
+    }
+    
+    
     public void seek(Term term) throws IOException {
       termDocs = ((IndexReader)fieldToReader.get(term.field()))
         .termPositions(term);
